presto-main/src/main/java/com/facebook/presto/connector/system/AbstractPropertiesSystemTable.java

```diff
@@ -80,6 +84,12 @@ public final RecordCursor cursor(ConnectorTransactionHandle transactionHandle, C
         Map<ConnectorId, Map<String, PropertyMetadata<?>>> connectorProperties = propertySupplier.get();
         for (Entry<String, ConnectorId> entry : new TreeMap<>(transactionManager.getCatalogNames(transactionId)).entrySet()) {
             String catalog = entry.getKey();
+            try {
+                accessControl.checkCanAccessCatalog(transactionId, session.getIdentity(), catalog);
+            }
+            catch (AccessDeniedException ignore) {
```

Comment:
it's usually a bad practice to do flow control with exceptions. Also do we want to really ignore the exception, why not propagate it?